node/src/tasks/user_submitted_file.rs

```diff
@@ -0,0 +1,111 @@
+use crate::services::file_transfer::commands::FileTransferServiceInterface;
+use crate::tasks::AcceptedBspVolunteer;
+use crate::tasks::StorageHubHandler;
+use crate::tasks::StorageHubHandlerConfig;
+use file_manager::traits::FileStorage;
+use log::{error, info};
+use sc_network::Multiaddr;
+use sc_network::PeerId;
+use sp_core::Blake2Hasher;
+use sp_core::Hasher;
+use std::str::FromStr;
+use storage_hub_infra::event_bus::EventHandler;
+use storage_hub_infra::types::Metadata;
+
+const LOG_TARGET: &str = "user-submitted-file-task";
+
+/// Handles the events related to users submitting files to be stored.
+pub struct UserSubmittedFileTask<SHC: StorageHubHandlerConfig> {
+    storage_hub_handler: StorageHubHandler<SHC>,
+}
+
+impl<SHC: StorageHubHandlerConfig> Clone for UserSubmittedFileTask<SHC> {
+    fn clone(&self) -> Self {
+        Self {
+            storage_hub_handler: self.storage_hub_handler.clone(),
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> UserSubmittedFileTask<SHC> {
+    pub fn new(storage_hub_handler: StorageHubHandler<SHC>) -> Self {
+        Self {
+            storage_hub_handler,
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> EventHandler<AcceptedBspVolunteer>
+    for UserSubmittedFileTask<SHC>
+{
+    async fn handle_event(&self, event: AcceptedBspVolunteer) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "Handling file submitted by user {:?} to BSP with location {:?}",
+            event.owner,
+            event.location,
+        );
+
+        // TODO: use `Multiaddr` instead of `String`.
+        let multiaddresses = event.multiaddresses;
+        let file_metadata = Metadata {
+            owner: event.owner.to_string(),
+            size: event.size.into(),
+            // TODO: use `FileLocation` instead of `String`.
+            location: format!("{:?}", event.location.into_inner()),
+            fingerprint: event.fingerprint,
+        };
+        let chunk_count = file_metadata.chunk_count();
+
+        // TODO(Arthur): double check this, I'm assuming Blake2 as the Trie hash function.
```

Comment:
We are not bound to use the same hash function as the Trie here but I agree we should double check.